py/obj.h

```diff
@@ -732,6 +732,12 @@ extern const struct _mp_obj_exception_t mp_const_GeneratorExit_obj;
 
 // General API for objects
 
+// Helper versions of m_new_obj when you need to immediately set base.type.
+// Implementing this as a call rather than inline saves 8 bytes per usage.
+#define m_new_obj_with_type(struct_type, obj_type) ((struct_type *)_m_new_obj_with_type(sizeof(struct_type), obj_type))
+#define m_new_obj_var_with_type(struct_type, var_type, var_num, obj_type) ((struct_type *)_m_new_obj_with_type(sizeof(struct_type) + sizeof(var_type) * (var_num), obj_type))
+void *_m_new_obj_with_type(size_t num_bytes, const mp_obj_type_t *type);
```

Comment:
Identifiers cannot start with `_`, that's reserved for the compiler/C runtime to use.

Also, do these belong in `misc.h` alongside the `m_new_obj`?

Reply:
Updated -- renamed to `mp_obj_malloc{,_var}` and the helper (in obj.c) is `mp_obj_malloc_helper`.